extensions/jackson/deployment/src/main/java/io/quarkus/jackson/deployment/JacksonProcessor.java

```diff
@@ -73,21 +78,40 @@ void register() {
             ignoredDotNames.add(ignoreJsonDeserializeClassBuildItem.getDotName());
         }
 
-        Collection<AnnotationInstance> pojoBuilderInstances = index.getAnnotations(JSON_DESERIALIZE);
-        for (AnnotationInstance pojoBuilderInstance : pojoBuilderInstances) {
-            if (AnnotationTarget.Kind.CLASS.equals(pojoBuilderInstance.target().kind())) {
-                DotName dotName = pojoBuilderInstance.target().asClass().name();
+        // handle the various @JsonDeserialize cases
+        for (AnnotationInstance deserializeInstance : index.getAnnotations(JSON_DESERIALIZE)) {
+            AnnotationTarget annotationTarget = deserializeInstance.target();
+            if (CLASS.equals(annotationTarget.kind())) {
+                DotName dotName = annotationTarget.asClass().name();
                 if (!ignoredDotNames.contains(dotName)) {
                     addReflectiveHierarchyClass(dotName);
                 }
 
-                AnnotationValue annotationValue = pojoBuilderInstance.value("builder");
+                AnnotationValue annotationValue = deserializeInstance.value("builder");
                 if (null != annotationValue && AnnotationValue.Kind.CLASS.equals(annotationValue.kind())) {
                     DotName builderClassName = annotationValue.asClass().name();
                     if (!BUILDER_VOID.equals(builderClassName)) {
                         addReflectiveHierarchyClass(builderClassName);
                     }
                 }
+            } else if (FIELD.equals(annotationTarget.kind()) || METHOD.equals(annotationTarget.kind())) {
+                AnnotationValue usingValue = deserializeInstance.value("using");
+                if (usingValue != null) {
+                    // the Deserializers are constructed internally by Jackson using a no-args constructor
+                    reflectiveClass.produce(new ReflectiveClassBuildItem(false, false, usingValue.asClass().toString()));
+                }
+            }
+        }
+
+        // handle the various @JsonSerialize cases
+        for (AnnotationInstance serializeInstance : index.getAnnotations(JSON_SERIALIZE)) {
```

Comment:
Why do we only add the class hierarchy for reflection in the deserialize case?

Reply:
We register the hierarchy when using the annotation on a field or method in either case.
The hierarchy comes into to play for deserialization when using `@JsonDeserialize` on a class which is a different case.
For the deserialization case we register the hierarchy when the annotation is applied to a Class.

Does that make sense?

Reply:
I suppose so :)

Reply:
I cleaned up what I wrote a little more for posterity :)